✨ feat: Add renameAccount method

Seroxdesign opened the pull request:
## Motivation and context

Renaming wallets is important for managing financials

Clearly and concisely describe the feature added/isses being solved.

## Does it fix any issue?

Adds functionality that isn't currently available in new dawn

#(issue)

## Other useful info

N/A

## Quality checklist

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough e2e tests.

**⚠️👆 Delete any section you see irrelevant before submitting the pull request 👆⚠️**

duckception commented:
Hey! Thanks for the PR! 🖤

It's perfect except for a few super small details 😇

duckception commented on wallets/metamask/test/e2e/metamask/renameAccount.spec.ts:
```diff
@@ -0,0 +1,30 @@
+test('should throw Zod error if an empty account name is passed', async ({ context, metamaskPage }) => {
+  const metamask = new MetaMask(context, metamaskPage, basicSetup.walletPassword)
+
+  await expect(metamask.renameAccount('')).rejects.toThrowError(z.ZodError)
+})
```
Make sure we check the exact error message here.

Seroxdesign replied:
Good catch

duckception commented on wallets/metamask/test/e2e/metamask/renameAccount.spec.ts:
```diff
@@ -0,0 +1,30 @@
+test('should throw Zod error if a reserved account name is passed', async ({ context, metamaskPage }) => {
+  const metamask = new MetaMask(context, metamaskPage, basicSetup.walletPassword)
+
+  await expect(metamask.renameAccount('Account 12')).rejects.toThrowError(z.ZodError)
```
Same here.

Seroxdesign replied:
Removed this because I removed validation

duckception commented on wallets/metamask/src/pages/HomePage/actions/renameAccount.ts:
```diff
@@ -0,0 +1,22 @@
+  await page.locator(Selectors.accountMenu.renameAccountMenu.listItemDetailButton).click()
+  await page.locator(Selectors.accountMenu.renameAccountMenu.renameButton).click()
+  await page.locator(Selectors.accountMenu.renameAccountMenu.renameInput).fill(newAccountName)
+  await page.keyboard.press('Enter')
```
This should be a mouse click instead 🫡

duckception commented on wallets/metamask/src/pages/HomePage/actions/renameAccount.ts:
```diff
@@ -0,0 +1,22 @@
+
+const Account = z.object({
+  name: z.string().refine((value) => {
+    return value.trim().length > 0 && !/^Account\s*\d+/.test(value);
```
We cannot verify this using the word `Account` because MetaMask can be used in different languages. That's why our code needs to be language-agnostic.

Even though I love this, I'm not sure what's the best way to handle this here. If you don't have any other idea, feel free to just drop this schema validation and add a `TODO` message:
`// TODO: See if we can add language-agnostic schema validation here.`

Seroxdesign commented:
> Hey! Thanks for the PR! 🖤
> 
> It's perfect except for a few super small details 😇

Thanks for taking the time to review, I pushed up a few changes that should make this more stable, specifically i removed the validation, made it so that I use a mouse click instead of an enter key, and removed a test spec I wrote for the English reserved names.

Good catches btw

Seroxdesign commented:
I'd appreciate it if we can rerun the failed tasks, I changed the title of the PR to fit the standard.

Seroxdesign commented:
I ran lint and redid the title, hopefully this is ready to go (:

duckception commented:
@drptbl Can you handle this? Looks decent 👍

Seroxdesign commented:
@drptbl can you please rerun this Validate Title check? 
Last one should've been valid, not sure why it failed. Any idea?

Seroxdesign commented:
@duckception @drptbl either of you able to get this merged?
The naming convention is being followed from what I understand but if not lmk what i'm doing wrong.
